extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -80,27 +78,15 @@ public OpenTelemetryMetricsProtobufReader(
   @Override
   public CloseableIterator<InputRow> read()
   {
-    Supplier<Iterator<InputRow>> supplier = Suppliers.memoize(() -> readAsList().iterator());
```

Comment:
it looks like we might want to keep the memoizing supplier here, since it was added to avoid the scenario where exceptions are thrown before we actually call next(), see https://github.com/confluentinc/druid/pull/99

As you had pointed out in that PR we should have added a test so that we wouldn't accidentally regress here.